easybuild/easyblocks/m/mesa.py

```diff
@@ -44,38 +45,61 @@ def __init__(self, *args, **kwargs):
-            # set cpu features of SWR for current architecture (only on x86_64)
-            if get_cpu_architecture() == X86_64:
-                feat_to_swrarch = {
-                    'avx': 'avx',
-                    'avx1.0': 'avx',  # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
-                    'avx2': 'avx2',
-                    'avx512f': 'skx',  # AVX-512 Foundation - introduced in Skylake
-                    'avx512er': 'knl',  # AVX-512 Exponential and Reciprocal Instructions implemented in Knights Landing
-                }
-                # determine list of values to pass to swr-arches configuration option
-                cpu_features = get_cpu_features()
-                self.swr_arches = sorted([swrarch for feat, swrarch in feat_to_swrarch.items() if feat in cpu_features])
+        self.gallium_configopts = []
+
+        # Check user-defined Gallium drivers
+        gallium_drivers = self.get_configopt_value('gallium-drivers')
+
+        if not gallium_drivers:
+            # Add appropriate Gallium drivers for current architecture
+            arch = get_cpu_architecture()
+            arch_gallium_drivers = {
+                'x86_64': ['swrast', 'swr'],
+                'POWER': ['swrast'],
+            }
+            if arch in arch_gallium_drivers:
+                gallium_drivers = arch_gallium_drivers[arch]
+                # Add configopt for additional Gallium drivers
+                self.gallium_configopts += ['-Dgallium-drivers=' + ','.join(gallium_drivers)]
+
+        # Check user-defined SWR arches
+        self.swr_arches = self.get_configopt_value('swr-arches')
+
+        if not self.swr_arches and 'swr' in gallium_drivers:
+            # Set cpu features of SWR for current micro-architecture
+            feat_to_swrarch = {
+                'avx': 'avx',
+                'avx1.0': 'avx',  # on macOS, AVX is indicated with 'avx1.0' rather than 'avx'
+                'avx2': 'avx2',
+                'avx512f': 'skx',  # AVX-512 Foundation - introduced in Skylake
+                'avx512er': 'knl',  # AVX-512 Exponential and Reciprocal Instructions implemented in Knights Landing
+            }
+            # Determine list of values to pass to swr-arches configuration option
+            cpu_features = get_cpu_features()
+            self.swr_arches = sorted([swrarch for feat, swrarch in feat_to_swrarch.items() if feat in cpu_features])
+            # Add configopt for additional SWR arches
+            self.gallium_configopts += ['-Dswr-arches=' + ','.join(self.swr_arches)]
+
+    def get_configopt_value(self, configopt_name):
+        """
+        Return list of values for the given configuration option in configopts
+        """
+        configopt_value = [opt.split('=')[1] for opt in self.cfg['configopts'].split() if configopt_name in opt]
+
+        if configopt_value:
+            configopt_value = configopt_value[0].replace("'", '').replace('"', '')
```

Comment:
@lexming Maybe make sure `configopt_value` has exactly one element here, to avoid surprises?

```python
if len(configopt_value) > 1:
    raise EasyBuildError("Found multiple values for %s in configopts: %s", ', '.join(configopt_value))

Reply:
good point, fixed :+1: